File: models/Map.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const MapSchema = new Schema({
    userId: {
        type: mongoose.Types.ObjectId,
        required: true
    },
    createdAt: {
        type: Date,
        default: new Date()
    },
    startTime: {
        type: Date,
        required: true
    },
    endTime: {
        type: Date,
        required: true
    },
    startLocation: {
        type: Object,
        required: true
    },
    endLocation: {
        type: Object,
        required: true
    },
    route: {
        type: Array,
        required: true
    },
    distance: {
        meters: {
            type: Number,
            default: 0.0
        },
        hr: {
            type: String,
            default: '0 km'
        }
    },
    duration: {
        seconds: {
            type: Number,
            default: 0
        },
        hr: {
            type: String,
            default: '0 min'
        }
    }
});

mongoose.model('maps', MapSchema);

File: app/src/main/java/com/swordrunner/swordrunner/data/model/Map.java
package com.swordrunner.swordrunner.data.model;

import com.google.gson.annotations.SerializedName;

import java.util.Date;
import java.util.List;

public class Map {

    @SerializedName("_id")
    public String id = null;

    @SerializedName("user_id")
    String userId = null;

    @SerializedName("created_at")
    public Date createdAt = null;

    @SerializedName("start_time")
    public Date startTime = null;

    @SerializedName("end_time")
    public Date endTime = null;

    @SerializedName("start_location")
    public TimeGeoPoint startLocation = null;

    @SerializedName("end_location")
    public TimeGeoPoint endLocation = null;

    @SerializedName("route")
    public List<TimeGeoPoint> route = null;

}